### ISSUE TITLE
Truncated Mean Percentage 

### ISSUE THREAD

**Issue by ktorktor** [markrogoyski/math-php#440]
Hi,

Thank you for making such a great math library!

I noticed a little error when calculating the Truncated Mean, when the percentage is set to 25% it actually calculates the truncated mean with a percentage of 50%

A quick fix would be to change line 510 in Average.php to:

`$trim_count = \floor($n * ($trim_percent/ 2/ 100));`

**Comment by markrogoyski** [comment]
Hi @ktorktor,

Thanks for your interest in MathPHP and your kind words.

Can you give an example input and expected output of where you think the library is miscalculating the truncated mean?

Thanks,
Mark

**Comment by markrogoyski** [comment]
Hi @ktorktor,

Thanks for providing an example.

Here is the result of your data set in three different math libraries: MathPHP, R, and NumPy/SciPy.

MathPHP
```php
php > $nums = [6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6];
php > echo \MathPHP\Statistics\Average::truncatedMean($nums, 50);
5
php > echo \MathPHP\Statistics\Average::truncatedMean($nums, 25);
4.7272727272727
```

R
```R
> nums <- c(6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6)
> mean(nums, trim=0.50)
[1] 5
> mean(nums, trim=0.25)
[1] 4.727273
```

Python (SciPy)
```python
In [5]: from scipy import stats

In [6]: nums = [6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6]

In [7]: stats.trim_mean(nums, 0.50)
Out[7]: 5.0

In [8]: stats.trim_mean(nums, 0.25)
Out[8]: 4.7272727272727275
```

All three libraries seem to agree on the answer. I also checked with this online calculator: https://www.easycalculation.com/statistics/trimmed-mean.php and it agreed with the results.

I think this is just an issue of semantics. Does 25% mean 25% from each side, or 25% of all numbers. It seems like it means for former for most programatic usages.

It looks like there is no issue with the calculation and is in alignment with other mathematics software libraries. However, I do think there is a bounds issue if you go above 50%. I'll update the code to provide a better error message.

Let me know if you have any other questions or comments about this. Thanks.
Mark

**Comment by markrogoyski** [comment]
Hi @ktorktor,

Thanks for providing an example.

Here is the result of your data set in three different math libraries: MathPHP, R, and NumPy/SciPy.

MathPHP
```php
php > $nums = [6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6];
php > echo \MathPHP\Statistics\Average::truncatedMean($nums, 50);
5
php > echo \MathPHP\Statistics\Average::truncatedMean($nums, 25);
4.7272727272727
```

R
```R
> nums <- c(6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6)
> mean(nums, trim=0.50)
[1] 5
> mean(nums, trim=0.25)
[1] 4.727273
```

Python (SciPy)
```python
In [5]: from scipy import stats

In [6]: nums = [6,4,2,4,3,7,6,33,77,22,3,5,6,5,0,2,3,4,6]

In [7]: stats.trim_mean(nums, 0.50)
Out[7]: 5.0

In [8]: stats.trim_mean(nums, 0.25)
Out[8]: 4.7272727272727275
```

All three libraries seem to agree on the answer. I also checked with this online calculator: https://www.easycalculation.com/statistics/trimmed-mean.php and it agreed with the results.

I think this is just an issue of semantics. Does 25% mean 25% from each side, or 25% of all numbers. It seems like it means for former for most programatic usages.

It looks like there is no issue with the calculation and is in alignment with other mathematics software libraries. However, I do think there is a bounds issue if you go above 50%. I'll update the code to provide a better error message.

Let me know if you have any other questions or comments about this. Thanks.
Mark

**Comment by ktorktor** [comment]
Thank you, Mark, I think this is just an issue of semantics as you said, there is nothing wrong with the calculation. I used TRIMMEAN from Google Sheets or Excel were from the same set of numbers as above.
 = TRIMMEAN(range,0.5) will result 4.73

It seems there isn't a well-established standard for this.  Maybe it helps to say in the comment that is a percentage cut from one end. In this case:

`@param int     $trim_percent Percent between 0-99`

should be

`@param int     $trim_percent Percent between 0-49`

**Comment by markrogoyski** [comment]
This has been addressed in the latest version [v2.6.0](https://github.com/markrogoyski/math-php/releases/tag/v2.6.0). Thanks again for reporting this issue.
